oxd-server/src/main/java/org/gluu/oxd/server/persistence/service/PersistenceServiceImpl.java

```diff
@@ -35,19 +38,30 @@ public void create() {
     }
 
     private PersistenceService createServiceInstance() {
-        String storage = configurationService.getConfiguration().getStorage();
-        if ("h2".equalsIgnoreCase(storage)) {
-            sqlProvider = new H2PersistenceProvider(configurationService);
-            setTimerForDBCleanUpTask();
-            return new SqlPersistenceServiceImpl(sqlProvider, configurationService);
-        } else if ("redis".equalsIgnoreCase(storage)) {
-            return new RedisPersistenceService(configurationService.getConfiguration());
-        } else if ("jdbc".equalsIgnoreCase(storage)) {
-            sqlProvider = new JDBCPersistenceProvider(configurationService);
-            setTimerForDBCleanUpTask();
-            return new SqlPersistenceServiceImpl(sqlProvider, configurationService);
+
+        String storage = this.configurationService.getConfiguration().getStorage();
+        switch (storage) {
+            case "h2":
+                this.sqlProvider = new H2PersistenceProvider(this.configurationService);
+                setTimerForDBCleanUpTask();
+                return new SqlPersistenceServiceImpl(this.sqlProvider, this.configurationService);
+            case "redis":
+                return new RedisPersistenceService(this.configurationService.getConfiguration());
+            case "jdbc":
+                this.sqlProvider = new JDBCPersistenceProvider(this.configurationService);
+                setTimerForDBCleanUpTask();
+                return new SqlPersistenceServiceImpl(this.sqlProvider, this.configurationService);
+            case "gluu_server_configuration":
+                setTimerForDBCleanUpTask();
+                return new GluuPersistenceService(this.configurationService.getConfiguration());
+            case "ldap":
+                setTimerForDBCleanUpTask();
+                return new GluuPersistenceService(this.configurationService.getConfiguration(), storage);
+            case "couchbase":
+                setTimerForDBCleanUpTask();
```

Comment:
That is wrong, we don't need timer for couchbase.
Add to expired entity
```
@Expiration
private Integer ttl; // in seconds
```
Just set expiration in seconds and couchbase will clean it up automatically.

Reply:
Done